lightning/src/ln/channelmanager.rs

```diff
@@ -2514,7 +2514,21 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>
 						}));
 					}
 				}
-				try_chan_entry!(self, chan.get_mut().update_add_htlc(&msg, pending_forward_info), channel_state, chan);
+        let create_onion_error_packet = |chan: &Channel<ChanSigner>, pending_forward_info: &PendingHTLCStatus| {
+          match pending_forward_info {
+            &PendingHTLCStatus::Forward(PendingHTLCInfo { ref incoming_shared_secret, .. }) => {
+              let chan_update = self.get_channel_update(chan).unwrap();
+              Some(onion_utils::build_first_hop_failure_packet(incoming_shared_secret, 0x1000|7, &{
```

Comment:
Not sure about this, but would "UPDATE|12 (fee_insufficient)" be more appropriate?

Reply:
I'm not certain, but I thought that error refers specifically to insufficient _routing_ fee. 

Further down in the onion-routing BOLT:
```
A forwarding node MAY, but a final node MUST NOT:
...
if the HTLC does NOT pay a sufficient fee:

    report the amount of the incoming HTLC and the current channel setting for the outgoing channel.
    return a fee_insufficient error.
```